version: '3.8'

services:
  app:
    build: .
    ports:
      - "8000:8000"
    restart: "on-failure"
    environment:
      REDIS_URL: redis://redis:6379/1
      USE_MOCK_EMBEDDINGS: "True"
    depends_on:
      - redis
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 3
  
  redis:
    image: "redis:7-alpine"
    volumes:
      - redis-data:/data
    ports:
      - "6379:6379"

  celery_worker:
    build: .
    restart: "on-failure"
    environment:
      REDIS_URL: redis://redis:6379/1
      USE_MOCK_EMBEDDINGS: "True"
    command: python -m celery -A src.celery_worker:celery_app worker --loglevel=info
    depends_on:
      - redis
      
volumes:
  redis-data:
